From 0b8bcbd3618bfb24f1552be7e251bef6a3eca411 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Mon, 2 Nov 2015 18:09:43 +0800
Subject: [PATCH 1022/1691] MLK-11809: watchdog: imx2_wdt: correct timeout
 after set_timeout()

commit 0b8bcbd3618bfb24f1552be7e251bef6a3eca411 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

Correct wdog->timeout after set_timeout(), otherwise, the WDIOC_GETTIMEOUT
will return the default value '60s' always, correct it and also for the new
wdog->pretimeout.

Signed-off-by: Robin Gong <b38343@freescale.com>
---
 drivers/watchdog/imx2_wdt.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/watchdog/imx2_wdt.c b/drivers/watchdog/imx2_wdt.c
index 5b9f9af..805df6b 100644
--- a/drivers/watchdog/imx2_wdt.c
+++ b/drivers/watchdog/imx2_wdt.c
@@ -181,6 +181,9 @@ static int imx2_wdt_set_timeout(struct watchdog_device *wdog,
 
 	regmap_update_bits(wdev->regmap, IMX2_WDT_WCR, IMX2_WDT_WCR_WT,
 			   WDOG_SEC_TO_COUNT(new_timeout));
+
+	wdog->timeout = new_timeout;
+
 	return 0;
 }
 
@@ -241,6 +244,8 @@ static int imx2_wdt_set_pretimeout(struct watchdog_device *wdog, unsigned int ne
 
 	regmap_write(wdev->regmap, IMX2_WDT_WICR, val);
 
+	wdog->pretimeout = new_timeout;
+
 	return 0;
 }
 
-- 
1.9.1

